### ISSUE TITLE
Introspectable alternative to $env.config.menus/keybindings

### ISSUE THREAD

**Issue by dyskos** [nushell/nushell#14682]
### Related problem

After upgrading from 0.100.0 -> 0.101.0, I decided to try out the new changes to the config. Most things went well: I was able to delete the original `config.nu`/`env.nu`, and include all my personal changes into a single `config.nu` without major problems.

However, I noticed that my color change for the `completion_menu` marker no longer worked.

Upon further research, I discovered that this is because `$env.config.menus` & `$env.config.keybindings` are no longer introspectable, and thus the only way to make a change is to overwrite the current settings with an entirely new record, like so:
```
$env.config.menus ++= [
    {
        name: completion_menu
        only_buffer_difference: false
        marker: $"(ansi white)| "
        type: {
            layout: columnar
            columns: 4
            col_width: 20
            col_padding: 2
        }
        style: {
            text: green
            selected_text: { attr: r }
            description_text: yellow
            match_text: { attr: u }
            selected_match_text: { attr: ur }
        }
    }
]
```

Needless to say, I'm not a fan of this, as before I only needed to do the following to make this minor change:
```
$env.config.menus.0.marker = $"(ansi white)| "
```

### Describe the solution you'd like

In [this](https://www.nushell.sh/blog/2024-12-04-configuration_preview.html) blog post, it is stated that "Only user-defined keybindings and menus should (as best-practice) be specified in the `$env.config`". I'm not sure why it should be the case that these things are only defined by the user, as many other values inside of `$env.config` are auto-populated by non-user-defined values.

But okay, let's instead say that  _`$env.config.menus`_ & _`$env.config.keybindings`_ should only be specified by the user. What if, on top of this, there were two more variables defined that are auto-populated by the internal defaults. That is, the user will be able to introspect (and modify) the defaults using the following new env variables:
- `$env.config.menus_internal`
- `$env.config.keybindings_internal`

For everyone already using custom, user-defined variables defined using `$env.config.menus ++= [ { name: ... } ]`, there would be no change.

For me, all I would have to do is change my above previously working solution from `$env.config.menus.0.marker = $"(ansi white)| "` to `$env.config.menus_internal.0.marker = $"(ansi white)| "`, and my new config will be able to remain clean and simple.

Of course, these extra variables are only necessary if `$env.config.menus` & `$env.config.keybindings` are really to be kept unpopulated by the internal defaults. If these are simply populated by the internal defaults, there would presumably be no breaking changes with anyone's current user menu/keybinding definitions, and my original config would continue to work as it did before.

### Describe alternatives you've considered

There are no real alternatives to this; right now I am forced to include an entire record of unchanged attributes just to change a single attribute, as opposed to my original method of altering a single attribute directly.

**Comment by fdncred** [comment]
ya, I agree. I believe we should populate keybindings and menus with the defaults so they show up with the $env command.

**Comment by NotTheDr01ds** [comment]
> However, I noticed that my color change for the `completion_menu` marker no longer worked.

Yes, this is definitely a gap and was [identified as a breaking change](https://www.nushell.sh/blog/2024-12-24-nushell_0_101_0.html#breaking-startup-config-changes-toc) in the release notes.

It's definitely something that should be fixed.

> What if, on top of this, there were two more variables defined that are auto-populated by the internal defaults. That is, the user will be able to introspect (and modify) the defaults using the following new env variables:
> 
> * `$env.config.menus_internal`
> * `$env.config.keybindings_internal`

Built-in keybindings are already introspectable via:

```
keybindings default
```

However, they aren't in a form that would be useful to (programmatically) turn into Nushell-usable values.  I'm not sure that this is a problem, since I don't think anyone is "appending" or merging values into an existing keybinding.  Is there a scenario in which this would be useful?

If so, then some form of the `keybindings` command should return a Nushell value for those.

For menus, we need something similar - Where internal defaults can be introspected and re-used for this purpose.

In the meantime, we've talked about adding some of the menus as constants in `std/config`.

The problem with this is the same as the problem with having them in `$env.config` (as we used to).  Without an *internal* way of introspecting these values, someone has to manually copy them over **and** keep them in sync.  The menus in the old `config.nu` were a mix of actual defaults as well as example menus.

So:

* Short-term: Have constants for them in `std/config`
* Long-term: A Nushell command that returns the actual default values of the menus.

**Comment by JustForFun88** [comment]
> Built-in keybindings are already introspectable via:
> 
> ```
> keybindings default
> ```
> 
> However, they aren't in a form that would be useful to (programmatically) turn into Nushell-usable values. I'm not sure that this is a problem, since I don't think anyone is "appending" or merging values into an existing keybinding. Is there a scenario in which this would be useful?


To be honest, the new settings system is a bit inconvenient. It would be really helpful if keybindings default provided bindings in a format that could be directly copied and pasted into the configuration.

This is particularly important for non-English keyboards when you're using multiple layouts simultaneously. Constantly switching back to the English layout to use keybindings is frustrating, as I might forget which layout I'm currently using. For example, I press `Ctrl + A`, but it doesn’t work because the layout is wrong. That’s why I immediately create additional keybindings for other layouts.

However, with the latest settings, this has become a pain because I don’t remember all the keybindings by heart, and the output of keybindings default is, frankly, useless.
